controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -156,12 +158,12 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 	}
 
 	// Handle imageRepository.
-	toImageRepository := fmt.Sprintf("%s/%s", parsedImage.Repository, parsedImage.Name)
+	toImageRepository := parsedImage.Repository
 	if clusterConfig.ImageRepository != "" {
-		toImageRepository = fmt.Sprintf("%s/%s", clusterConfig.ImageRepository, coreDNSKey)
+		toImageRepository = strings.TrimSuffix(clusterConfig.ImageRepository, "/")
 	}
 	if clusterConfig.DNS.ImageRepository != "" {
-		toImageRepository = fmt.Sprintf("%s/%s", clusterConfig.DNS.ImageRepository, coreDNSKey)
+		toImageRepository = strings.TrimSuffix(clusterConfig.DNS.ImageRepository, "/")
```

Comment:
Do we need the trim suffix here?

Reply:
given that the image repository is provider by the user, and that down we are concatenating it with image name and tag,   I thought that removing the suffix makes the concat operation more robust (it will work with the user providing "k8s.gcr.io" but also "k8s.gcr.io/")

Reply:
(see comment above)